driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocumentCodec.java

```diff
@@ -44,6 +46,7 @@
     ChangeStreamDocumentCodec(final Class<TResult> fullDocumentClass, final CodecRegistry codecRegistry) {
 
         ClassModelBuilder<ChangeStreamDocument> classModelBuilder = ClassModel.builder(ChangeStreamDocument.class);
+        ((PropertyModelBuilder<BsonTimestamp>) classModelBuilder.getProperty("clusterTime")).codec(new BsonTimestampCodec());
```

Comment:
Tests all passed even without this line, but added it anyway for consistency.

Reply:
With the default registry its fine - I think these guards are to protect against non standard registries.